ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/core/TransactionPool.java

```diff
@@ -78,6 +81,12 @@ public TransactionPool(
   }
 
   public void addRemoteTransactions(final Collection<Transaction> transactions) {
+
+    if (synchronizer.getSyncStatus().isPresent()) {
```

Comment:
This is too aggressive.  If we are behind by as much as 1 block we are going to drop transactions.  This can happen from the p2p network's normal operation.  We need a threshold, stored in the SynchronizerConfiguration, such as 100.  If we are syncing and more than that threshold behind then we can drop results.  But if we are not syncing or are within the threshold we should hold onto it.